Shared/Action/TelemetryAction.swift

```diff
@@ -61,6 +62,32 @@ enum ExtraKey: String {
     case fxauid, itemid, error
 }
 
+class GleanActionHandler: ActionHandler {
+    private let glean = Glean.shared
+    
+    private var disposeBag = DisposeBag()
+    
+    init() {
+        UserDefaultStore.shared.recordUsageData.subscribe(
+            onNext: { uploadEnabled in
+                self.glean.setUploadEnabled(uploadEnabled)
+            },
+            onError: nil,
+            onCompleted: nil,
+            onDisposed: nil
+        ).disposed(by: self.disposeBag)
+        
+        glean.initialize()
+    }
+    
+    lazy var telemetryActionListener: AnyObserver<TelemetryAction> = {
```

Comment:
Do we need to declare this here? Can we move this up to a follow-up PR in which we add events?

Reply:
I first thought it would be necessary to tie the preference to `setUploadEnabled` but it wasn't.  I can strip this out for now and add it when I instrument the events.